nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,281 @@
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/db.sqlite3"
+                '';
+              };
+              type = lib.mkOption {
+                type = lib.types.enum [ "sqlite" ];
+                default = "sqlite";
+                description = "Database type.";
+              };
+            };
+            server = {
+              address = lib.mkOption {
+                type = lib.types.str;
+                default = "127.0.0.1:${toString cfg.port}";
+                description = "Server address.";
+                defaultText = lib.literalExpression ''
+                  "127.0.0.1:''${config.services.omnom.port}"
+                '';
+              };
+              secure_cookie = lib.mkOption {
+                type = lib.types.bool;
+                default = true;
+                description = "Whether to limit cookies to a secure channel.";
+              };
+            };
+            storage = {
+              type = lib.mkOption {
+                type = lib.types.str;
+                default = "fs";
+                description = "Storage type.";
+              };
+              root = lib.mkOption {
+                type = lib.types.path;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+            smtp = {
+              tls = lib.mkEnableOption "Whether TLS encryption should be used.";
+              tls_allow_insecure = lib.mkEnableOption "Whether to allow insecure TLS.";
+              host = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "SMTP server hostname.";
+              };
+              port = lib.mkOption {
+                type = lib.types.port;
+                default = 25;
+                description = "SMTP server port address.";
+              };
+              sender = lib.mkOption {
+                type = lib.types.str;
+                default = "Omnom <omnom@127.0.0.1>";
+                description = "Omnom sender e-mail.";
+              };
+              send_timeout = lib.mkOption {
+                type = lib.types.int;
+                default = 10;
+                description = "Send timeout duration in seconds.";
+              };
+              connection_timeout = lib.mkOption {
+                type = lib.types.int;
+                default = 5;
+                description = "Connection timeout duration in seconds.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !lib.hasAttr "username" cfg.settings.smtp;
```

Comment:
I don't think it's necessary to place the username in a secrets file, also we do this nowhere else in nixpkgs.

Reply:
Out of curiosity, don't you think it's a better practice to not store usernames in cleartext?

> we do this nowhere else in nixpkgs

Doesn't mean we shouldn't innovate and improve the status quo ;)

Reply:
I don't think that the username is a secret.